src/main/java/core/mate/academy/service/MachineService.java

```diff
@@ -17,7 +17,7 @@
      * @param type - any class of Machine sub class. For example: Track.class or Bulldozer.class
      * @return the list of machines
      */
-    List<Machine> getAll(Class type);
+    List<Machine> getAll(Class<? extends Machine> type);
```

Comment:
let's use T instead of Machine with all methods in this interface

Reply:
done